kernel/device.c

```diff
@@ -86,15 +89,19 @@ const struct device *z_impl_device_get_binding(const char *name)
 	 */
 	for (dev = __device_start; dev != __device_end; dev++) {
 		if (z_device_ready(dev) && (dev->name == name)) {
+			LOG_DBG("%s found", name);
 			return dev;
 		}
 	}
 
 	for (dev = __device_start; dev != __device_end; dev++) {
+		LOG_DBG("   - %s, ready=%d", dev->name, z_device_ready(dev));
 		if (z_device_ready(dev) && (strcmp(name, dev->name) == 0)) {
+			LOG_DBG("%s found", name);
 			return dev;
 		}
 	}
+	LOG_DBG("%s not found", name);
```

Comment:
In fact only that logging "would" be fine. Rest is useless, if it's found everything is fine.

Reply:
Well if not found, how do you find out what went wrong? The point of the logging is to show you.

Or are you suggesting to do one more search with logging enabled?